### DISCUSSION TITLE
请问能够在一个项目中同时支持Mysql和Sqlite吗？

### DISCUSSION THREAD

**Discussion by TreatTrick** [qicosmos/ormpp#184] [Q&A]
如题，我同时打开了ORMPP_ENABLE_MYSQL和ORMPP_ENABLE_SQLITE3后，有一些语句会出错，是不是目前还不能支持在一个项目中同时使用mysql和sqlite3

**Comment by Jacyking** [comment]
用cmake分别生成两个lib
建议加qq群

**Reply by qicosmos** [reply]
把具体的解决方法贴到这里，群里讨论后面就难查找记录了。

**Reply by TreatTrick** [reply]
请问具体的方法是什么呢？我试了一下，但是因为模板类在CMake中只能add_library(ormpp INTERFACE)，这个INTERFACE会导致我的宏定义传播到所有引用他的库里，所以我定义了两个库分别开了ORMPP_ENABLE_MYSQL和ORMPP_ENABLE_SQLITE3的宏定义，但是最后引用我这两个库的库还是会同时有上述两个宏定义，导致编译出来的代码还是有问题。
请问是不是应该用其他的什么方式？
